src/notation/inotationinteraction.h

```diff
@@ -266,6 +268,8 @@ class INotationInteraction
 
     virtual void showItem(const Ms::EngravingItem* item, int staffIndex = -1) = 0;
     virtual async::Channel<ShowItemRequest> showItemRequested() const = 0;
+
+    virtual ScoreCallbacks* scoreCallbacks() = 0;
```

Comment:
We should not expose ScoreCallbacks (and onElementDestruction) in this interface

Reply:
I would suggest replacing MuseScoreView::onElementDestruction with async::Channel<EngravingItem*> MuseScoreView::elementDestroyed(). After this change, we will be able to connect to the channel where needed and update all local pointers to avoid crashes

Reply:
> We should not expose ScoreCallbacks (and onElementDestruction) in this interface

We need to use it in NotationNoteInput. It looks like the MuseScoreView interface was designed to have only one instance of it for each notation, so I would prefer if NotationNoteInput reuses the ScoreCallbacks from NotationInteraction. 

Ultimately I think the whole MuseScoreView interface needs some rethinking. We could split it into smaller groups of related callbacks, and maybe move some functionality to the score itself (for example, the `elementDestroyed` notification could be part of the score class).